source/posix/thread.c

```diff
@@ -208,6 +224,33 @@ int aws_thread_launch(
         goto cleanup;
     }
 
+#if AWS_AFFINITY_METHOD == AWS_AFFINITY_METHOD_PTHREAD
+    /* If we don't have pthread_attr_setaffinity_np, we may
+     * still be able to set the thread affinity after creation. */
+    if (options && options->cpu_id >= 0) {
+        AWS_LOGF_INFO(
+            AWS_LS_COMMON_THREAD,
+            "id=%p: cpu affinity of cpu_id %d was specified, attempting to honor the value.",
+            (void *)thread,
+            options->cpu_id);
+
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET((uint32_t)options->cpu_id, &cpuset);
```

Comment:
This is the same as above, you could move this out of the #if blocks.

Reply:
It needs to be guarded somehow though, I think because not all platforms have support for CPU_ZERO/SET. This was previously handled by the
```cpp
#if !defined(__MACH__) && !defined(__ANDROID__)
```

Reply:
Ah good point.